Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -431,14 +431,46 @@
     // FIXME: We should also set the openInPreviewIndex when UnifiedPdfPlugin::openWithPreview is implemented
     menuItems.append({ WebCore::contextMenuItemPDFOpenWithPreview(), true, false, 0, true, static_cast<uint8_t>(ContextMenuItemTag::OpenWithPreview) });
 
+    auto currentDisplayMode = contextMenuItemTagFromDisplyMode(m_documentLayout.displayMode());
+    // Single page
+    menuItems.append({ WebCore::contextMenuItemPDFSinglePage(), true, false, currentDisplayMode == ContextMenuItemTag::SinglePage, true, static_cast<uint8_t>(ContextMenuItemTag::SinglePage) });
+
+    // Single page continous
+    menuItems.append({ WebCore::contextMenuItemPDFSinglePageContinuous(), true, false, currentDisplayMode == ContextMenuItemTag::SinglePageContinuous, true, static_cast<uint8_t>(ContextMenuItemTag::SinglePageContinuous) });
+
+    // Two pages
+    menuItems.append({ WebCore::contextMenuItemPDFTwoPages(), true, false, currentDisplayMode == ContextMenuItemTag::TwoPages, true, static_cast<uint8_t>(ContextMenuItemTag::TwoPages) });
+
+    // Two pages continuous
+    menuItems.append({ WebCore::contextMenuItemPDFTwoPagesContinuous(), true, false, currentDisplayMode == ContextMenuItemTag::TwoPagesContinuous, true, static_cast<uint8_t>(ContextMenuItemTag::TwoPagesContinuous) });
+
     return { contextMenuPoint, WTFMove(menuItems), { } };
 }
 
-void UnifiedPDFPlugin::performContextMenuAction(ContextMenuItemTag tag) const
+void UnifiedPDFPlugin::performContextMenuAction(ContextMenuItemTag tag)
 {
     switch (tag) {
     // The OpenWithPreviewAction is handled in the UI Process
     case ContextMenuItemTag::OpenWithPreview: return;
+    case ContextMenuItemTag::SinglePage:
+    case ContextMenuItemTag::SinglePageContinuous:
+    case ContextMenuItemTag::TwoPagesContinuous:
+    case ContextMenuItemTag::TwoPages:
+        if (tag != contextMenuItemTagFromDisplyMode(m_documentLayout.displayMode())) {
+            auto newDisplayMode = [&] {
+                switch (tag) {
+                case ContextMenuItemTag::SinglePage: return PDFDocumentLayout::DisplayMode::SinglePage;
```

Comment:
You have conversations between ContextMenuItemTag and PDFDocumentLayout::DisplayMode in two places. It might be nicer to have helper functions for doing the conversions both directions, next to each other in the source.

Reply:
Done